maltoolbox/language/languagegraph.py

```diff
@@ -26,44 +26,58 @@
 
 logger = logging.getLogger(__name__)
 
+def get_asset_name_from_full_attack_step(
+        attack_step_full_name: str) -> str:
+    return attack_step_full_name.split(':')[0]
+
+def get_attack_step_name_from_full_attack_step(
+        attack_step_full_name: str) -> str:
+    return attack_step_full_name.split(':')[1]
+
 @dataclass
 class LanguageGraphAsset:
-    name: Optional[str] = None
-    associations: list[LanguageGraphAssociation] = field(default_factory=lambda: [])
-    attack_steps: list[LanguageGraphAttackStep] = field(default_factory=lambda: [])
-    description: dict = field(default_factory=lambda: {})
-    # MAL languages currently do not support multiple inheritance, but this is
-    # futureproofing at its most hopeful.
-    super_assets: list = field(default_factory=lambda: [])
-    sub_assets: list = field(default_factory=lambda: [])
+    name: str
+    associations: dict[str, LanguageGraphAssociation] = \
+        field(default_factory = dict)
+    attack_steps: dict[str, LanguageGraphAttackStep] = \
+        field(default_factory = dict)
+    info: dict = field(default_factory = dict)
+    super_asset: Optional[LanguageGraphAsset] = None
+    sub_asset: Optional[LanguageGraphAsset] = None
+    variables: dict = field(default_factory = dict)
     is_abstract: Optional[bool] = None
 
     def to_dict(self) -> dict:
         """Convert LanguageGraphAsset to dictionary"""
         node_dict: dict[str, Any] = {
             'name': self.name,
-            'associations': [],
-            'attack_steps': [],
-            'description': self.description,
-            'super_assets': [],
-            'sub_assets': []
+            'associations': {},
+            'attack_steps': {},
+            'info': self.info,
+            'super_asset': self.super_asset.name if self.super_asset else "",
+            'sub_asset': self.sub_asset.name if self.sub_asset else "",
```

Comment:
```suggestion
            'super_asset': self.super_asset.name or "",
            'sub_asset': self.sub_asset.name or "",
```

Reply:
If it is None, won't the check `if self.sub_asset` fail and `else` will happen? Which is the same with `self.sub_asset.name or ""` where `self.sub_asset.name` will be falsy and the second element in the `or` expression will be chosen.

Reply:
Let me clarify this, `self.sub_asset` is missing, not `self.sub_asset.name`.

Reply:
ah, I see, I didn't read that code carefully.